### PR TITLE
Disable integration tests

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/elastic/package-registry/issues/698
Closes: https://github.com/elastic/package-registry/pull/691

This PR disables integration tests in the Package Registry repository. It removes all logic related to it. I have also removed the vendor directory to be consistent with other repositories.

### CODE REVIEW

File: testing/environments/local.yml [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-    depends_on:
-      elasticsearch:
-        condition: service_healthy
-
-  elasticsearch:
-    ports:
-      - "127.0.0.1:9200:9200"
-
-  package-registry:
-    build: ../..
```
Comment (jsoriano) [inline testing/environments/local.yml:19]: Is there still some test or something that tries to build the registry container?

File: magefile.go [hunk 2]
```diff
@@ -171,23 +149,3 @@ func Clean() error {
...
 	return os.RemoveAll("package-registry")
 }
-
-func Vendor() error {
-	fmt.Println(">> mod - updating vendor directory")
-
-	err := sh.RunV("go", "mod", "tidy")
-	if err != nil {
-		return err
-	}
```
Comment (jsoriano) [inline magefile.go:181]: We probably still want to check that `go mod tidy` doesn't produce changes.

**Review by jsoriano — COMMENTED** [review]
> I have also removed the vendor directory to be consistent with other repositories.

:+1:

File: testing/environments/local.yml (hunk 1 above)
Comment (mtojek) [reply testing/environments/local.yml:19]: Currently, it's only in the Jenkinsfile: https://github.com/elastic/package-registry/blob/master/.ci/Jenkinsfile#L164

File: magefile.go (hunk 2 above)
Comment (mtojek) [reply magefile.go:181]: Fixed, thanks

File: testing/environments/local.yml (hunk 1 above)
Comment (jsoriano) [reply testing/environments/local.yml:19]: I think this is enough, I only wanted to check that there is something checking that the container image can be built.

**Review by jsoriano — APPROVED** [review]
:sparkles: I love when code can be removed :slightly_smiling_face:

File: NOTICE.txt [hunk 3]
```diff
@@ -1,5 +0,0 @@
-Elastic package-registry
```
Comment (ruflin) [inline NOTICE.txt:1]: Is this file removed on purpose?

**Review by ruflin — COMMENTED** [review]
* Can we separate the changes around moving vendor directories and removing integration testing? 
* For the compose files: What is the new recommended way to test a registry change during development?

File: NOTICE.txt (hunk 3 above)
Comment (mtojek) [reply NOTICE.txt:1]: Yes, as there is also LICENSE file: https://github.com/elastic/package-registry/blob/master/LICENSE.txt

Comment (ruflin) [reply NOTICE.txt:1]: For example in Beats repo we have both files and I remember there was a reason for this. Not sure if we still need both? @urso you might remember?

**Comment by mtojek** [comment]
> Can we separate the changes around moving vendor directories and removing integration testing?

Ok, I can do this.

> For the compose files: What is the new recommended way to test a registry change during development?

It's the same as it was before, as integration tests were redundant. There are endpoint tests defined here: https://github.com/mtojek/package-registry/blob/disable-integration-tests/main_test.go#L47-L67

File: NOTICE.txt (hunk 3 above)
Comment (mtojek) [reply NOTICE.txt:1]: I see, let's clarify if we need this file. It isn't present in any package related repo (integrations, elastic-package, package-spec, etc.)

**Comment by mtojek** [comment]
I extracted the vendor part, so this PR went down to 11 files. @ruflin what do you think should be included in this PR to get it into the master branch?

**Comment by mtojek** [comment]
> For the testing and compose file, I was referring to manual testing. Lets say I have a PR and I want to check it quickly with an "actual" Kibana setup. Before I used the compose files to build the container and spin it up. As the compose files are not in anymore, does something similar still exist? I personally like to provide devs an easy way to manually test their changes.

I used to start it with "go run ." and then point Kibana to the local EPR instance.

**Comment by ruflin** [comment]
I assume in the above scenario, you run Kibana from source? How do you run it from a released version without having to setup all the stack manually?

**Comment by mtojek** [comment]
I think that it isn't necessary to run against the Kibana from source. You can docker-build the package-registry locally (with packages in the `build` directory) and then docker-tag it accordingly.

**Pull request merged**